bundles/org.openhab.binding.nanoleaf/src/main/java/org/openhab/binding/nanoleaf/internal/handler/NanoleafControllerHandler.java

```diff
@@ -426,38 +467,160 @@ private void runPanelDiscovery() {
         }
     }
 
+    /**
+     * This is based on the touch event detection described in https://forum.nanoleaf.me/docs/openapi#_842h3097vbgq
+     */
+    private static boolean touchJobRunning = false;
+    private void runTouchDetection() {
+        if (touchJobRunning) {
+            logger.debug("touch job already running. quitting.");
+            return;
+        }
+        try {
+            touchJobRunning = true;
+            URI eventUri = OpenAPIUtils.getUri(getControllerConfig(),API_EVENTS,"id=4");
+            logger.debug("touch job registered on: {}",eventUri.toString());
+            httpClient .newRequest(eventUri)
+                    .send(new Response.Listener.Adapter()  // request runs forever
+                    {
+                        @Override
+                        public void onContent(@Nullable Response response, @Nullable ByteBuffer content) {
+                            String s = StandardCharsets.UTF_8.decode(content).toString();
+                            logger.trace("content {}",s);
+
+                            Scanner eventContent =  new Scanner(s);
+                            while (eventContent.hasNextLine()) {
+                                String line = eventContent.nextLine().trim();
+                                // we don't expect anything than content id:4, so we do not check that but only care about the data part
+                                if (line.startsWith("data:")) {
+                                    String json = line.substring(5).trim();  // supposed to be JSON
+                                    try {
+                                        @Nullable TouchEvents touchEvents = gson.fromJson(json, TouchEvents.class);
+                                        handleTouchEvents(touchEvents);
+                                    } catch (JsonSyntaxException jse) {
+                                        logger.error("couldn't parse touch event json {}", json);

```

Comment:
In [E.4 of the coding guidelines](http://docs.openhab.org/developers/development/guidelines.html#e-logging) we listed what kind of logging level should be used when. Error should only be used:

> `error` logging should only be used to inform the user that something is tremendously wrong in his setup, the system cannot function normally anymore, and there is a need for immediate action. 

Please change this to use the correct logging levels and if possible change the ONLINE/OFFLINE status of thing / bridge.

Check all